internal/commands/document.go

```diff
@@ -79,21 +73,19 @@ func getPolicyDocumentation(path string, outputDirectory string) (string, error)
 	policyDocument += "|---|---|---|---|---|\n"
 
 	for _, policy := range policies {
-		var policyCommentBlocks []PolicyCommentBlock
 		policyCommentBlocks, err := getPolicyCommentBlocks(policy.Comments)
 		if err != nil {
 			return "", fmt.Errorf("get policy comment blocks: %w", err)
 		}
 
-		relPath, err := filepath.Rel(outputDirectory, policy.FilePath)
-		if err != nil {
-			return "", fmt.Errorf("rel path: %w", err)
-		}
-
-		relDir := filepath.Dir(relPath)
-		ruleTypes := strings.Join(policy.RulesActions, ", ")
-
 		for _, policyCommentBlock := range policyCommentBlocks {
+			relPath, err := filepath.Rel(outputDirectory, policy.FilePath)
+			if err != nil {
+				return "", fmt.Errorf("rel path: %w", err)
+			}
+
+			relDir := filepath.Dir(relPath)
+			ruleTypes := strings.Join(policy.RulesActions, ", ")
```

Comment:
Why was this moved inside the policyCommentBlocks iterator? This won't change until the next policy so we're potentially computing these more often than needed.

Reply:
It was mostly a readability play. I agree it's excess compute, but it feels wrong to keep them so far away from their use. I'm not strongly opinionated about it either way. This is probably a smell that this construction needs some refactoring love so things aren't so far apart from one another.

Reply:
I think this opens the larger discussion of should we allow/support more than one comment block per policy? Our tooling is really built around having a bunch of small policy files that check for one thing, even if that may be split out into multiple rules. If we didn't have to iterate over multiple policy blocks, that would resolve this as well.

Reply:
I'm more inclined to leans towards the idea of one comment block per policy file. It seems a little silly to support different `Kinds` for different `violations` given the same policy file. Smells of needing to split the file up. So if we only supported a single set of Kind matchers in a file, you could argue that means a single comment block as well.

The only counterexample I can think of at the moment is if you wanted to be able to keep the comment block with the violation itself--and the documentation would then be summing up all of those blocks:

```rego
// All containers must do stuff
// Here is an example of a valid container
// ..yaml..
// Yeah that's pretty valid, amirite?
// Kinds @v1/Pod

// Containers must set a memory limit.
// (you'd probably change this to have it make way more sense as a cohesive document)
violation[] {}

// Containers must set a request limit.
violation[] {}
```

doc would be (something like)

```markdown
All containers must do stuff
Here is an example of a valid container
..yaml..
Yeah that's pretty valid, amirite?

Containers must set a memory limit.

Containers must set a request limit.
```

All that said, it's more of something to consider as something that might pop up--but at least for now I'd rather optimize for the case of a single block.

Reply:
Agreed. I think we can merge this as it is and open a separate PR for addressing this issue.